crates/common/src/contracts.rs

```diff
@@ -95,25 +96,24 @@ pub type ContractsByAddress = BTreeMap<Address, (String, JsonAbi)>;
 
 /// Very simple fuzzy matching of contract bytecode.
 ///
-/// Will fail for small contracts that are essentially all immutable variables.
-pub fn diff_score(a: &[u8], b: &[u8]) -> f64 {
-    let max_len = usize::max(a.len(), b.len());
-    let min_len = usize::min(a.len(), b.len());
+/// Returns a value between `0.0` (identical) and `1.0` (completely different).
+pub fn bytecode_diff_score<'a>(mut a: &'a [u8], mut b: &'a [u8]) -> f64 {
+    if a.len() < b.len() {
+        std::mem::swap(&mut a, &mut b);
+    }
+
+    // Account for different lengths.
+    let mut n_different_bytes = a.len() - b.len();
 
-    if max_len == 0 {
+    // Skip checking individual bytes if the lengths are too different.
+    if n_different_bytes > 32 && n_different_bytes * 10 > a.len() {
```

Comment:
would pref if we moved the constants out and named them to make it clearer why we chose these numbers

Reply:
Added a comment, LMK